### ISSUE TITLE
Add truncation arguments to univariate distributions

### ISSUE THREAD

**Issue by joethorley** [poissonconsulting/extras#46]
```
log_lik_norm(x, mean, sd, tlower = -Inf, tupper = -Inf)
res_norm(x, mean, sd, tlower = -Inf, tupper = -Inf)
ran_norm(x, mean, sd, tlower = -Inf, tupper = -Inf)
```

if outside limits then log_lik = -Inf
res I guess is -Inf or Inf
and ran_norm should resample.

**Comment by joethorley** [comment]
the ran_norm might be problematic if narrow range cos either very few not NA or repeats many times.

**Comment by joethorley** [comment]
This is a stab at one

```
ran_student2 <- function(n = 1, mean = 0, sd = 1, theta = 0, tlower = -Inf, tupper = Inf) {
  chk_number(tlower)
  chk_number(tupper)
  chk_lt(tlower, tupper)

  ran <- extras::ran_student(n, mean, sd, theta = theta)
  ran <- ran[ran > tlower & ran < tupper]
  nran <- length(ran)

  if(nran == n) {
    return(ran)
  }

  ran %>%
    c(ran_student2(n - nran, mean = mean, sd = sd, theta = theta, tlower, tupper))
}
```

**Comment by joethorley** [comment]
Might not need to check proportion of ran as just get stack overflow.
Or perhaps check to give better error message.

**Comment by nehill197** [comment]
- This issue is more complex than we initially thought. 
- The log likelihoods need to incorporate the additional density from the truncated sections such the area under the PDF sums to 1.
- This means we need either distribution specific truncated PDFs or perhaps a general solution to incorporate the truncated density. 
- for Normal dist. see wikipedia page for PDF; `msm` package has `dtnorm()` and `rtnorm()` for comparison.

**Comment by joethorley** [comment]
Yes I think we are going to need to use the equivalent of pnorm and qnorm for each of our functions.

I guess we could call them something like prob_norm and quant_norm(). These should also have tlower and tupper arguments.

**Comment by nehill197** [comment]
Potentially useful information in the stan reference manual section 7.4
https://mc-stan.org/docs/reference-manual/sampling-statements.html

**Comment by nehill197** [comment]
@joethorley If I have time I may take a crack at this during the hackathon to at least get us log-likelihood functions for truncated distributions so that we can use them for the `lprior` component of the priorsense  sensitivity analyses. 

Do you have any thoughts on whether we should add the `tlower` and `tupper` arguments to the existing `log_lik_xxx()` functions, or add new functions like `log_lik_xxx2()` that have the truncation arguments?

**Comment by joethorley** [comment]
My leaning is to add to the existing functions as the defaults shouldn't change the behaviour.
